store/tikv/latch/scheduler.go

```diff
@@ -92,16 +92,19 @@ func (scheduler *LatchesScheduler) Close() {
 // Lock acquire the lock for transaction with startTS and keys. The caller goroutine
 // would be blocked if the lock can't be obtained now. When this function returns,
 // the lock state would be either success or stale(call lock.IsStale)
-func (scheduler *LatchesScheduler) Lock(startTS uint64, keys [][]byte) *Lock {
+func (scheduler *LatchesScheduler) Lock(startTS uint64, keys [][]byte) (*Lock, time.Duration) {
```

Comment:
This is too intrusive to the API.
It's very appreciable if you could add the metrics to the caller? like this:
```
start := time.Now()
sched.Lock(..)
waitTime = time.Since(start)
```

Reply:
yes, but let it in outer will call `time.Now` event no conflict just take care performance, ok it's move it out, @jackysp also agree with your point